src/engine/sys/sdl_glimp.cpp

```diff
@@ -1422,6 +1425,82 @@ static void GLimp_CheckGLEW( const glConfiguration &requestedConfiguration )
 	}
 }
 
+static bool IsSdlVideoRestartNeeded()
+{
+	if ( !firstSdlVideoAttempt )
+	{
+		return false;
+	}
+
+	/* We call RV600 the first generation of R600 cards, to make a difference
+	with RV700 and RV800 cards that are also supported by the Mesa R600 driver.
+
+	The Mesa R600 driver has broken Hyper-Z wth RV600, not RV700 neither RV800. */
+	static bool r600_hyperz_modified = false;
+	static const char* r600_hyperz = nullptr;
```

Comment:
This could use a comment or more specific naming. It took me a long time to figure out that the point is to restore a possible user-specified value of the environment variable.

But actually I might suggest that if there is any value already set for this, the workaround should be disabled.

Reply:
> But actually I might suggest that if there is any value already set for this, the workaround should be disabled.

I implemented that, which simplifies the code.  It also makes possible to remove `firstSdlVideoAttempt` as the presence of the cvar being set by the workaround is enough to know the restart already happened.

I have to test the code again on the hardware to be sure I have not introduced any regression from my original code.

Reply:
> I have to test the code again on the hardware to be sure I have not introduced any regression from my original code.

It still works, I haven't introduced any regression when simplifying the code.